cornice/validators/_colander.py

```diff
@@ -0,0 +1,70 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+from webob.multidict import MultiDict
+from colander import Invalid
+
+
+class SchemaError(Exception):
+    pass
+
+
+def extract_cstruct(request):
```

Comment:
Note: this could even go to `cornice.validators` since it does not contain any colander specificities. Plus, it would be useful for plugin custom validation libs IMO